graphql/batch_executor.go

```diff
@@ -16,11 +16,11 @@ import (
 // graphql query.
 type WorkUnit struct {
 	Ctx          context.Context
-	field        *Field
+	Field        *Field
```

Comment:
I am not comfortable to expose these field since the caller will be able to edit them. 
How about we add corresponding methods on WorkUnit struct?

Reply:
That can work!